codebase2/codebase-sqlite/U/Codebase/Sqlite/Queries.hs

```diff
@@ -1047,3 +1405,78 @@ ancestorSql =
       )
     SELECT * FROM ancestor
   |]
+
+-- * share sync / temp entities
+
+getMissingDependencyJwtsForTempEntity :: Base32Hex -> Transaction (Maybe (NESet Text))
+getMissingDependencyJwtsForTempEntity h = do
+  jwts <-
+    queryListCol
+      [here|
+        SELECT dependencyJwt FROM temp_entity_missing_dependency
+        WHERE dependent = ?
+      |]
+      (Only h)
+  pure (NESet.nonEmptySet (Set.fromList jwts))
+
+getMissingDependentsForTempEntity :: Base32Hex -> Transaction [Base32Hex]
+getMissingDependentsForTempEntity h =
+  queryListCol
+    [here|
+      SELECT dependent
+      FROM temp_entity_missing_dependency
+      WHERE dependency = ?
+    |]
+    (Only h)
+
+-- | Insert a new `temp_entity` row, and its associated 1+ `temp_entity_missing_dependency` rows.
+--
+-- Preconditions:
+--   1. The entity does not already exist in "main" storage (`object` / `causal`)
+--   2. The entity does not already exist in `temp_entity`.
```

Comment:
Does this need to be a pre-condition or could we just check and bail if it already exists?